src/hotspot/share/gc/g1/g1YoungGCEvacFailureInjector.cpp

```diff
@@ -30,6 +30,33 @@
 
 #if EVAC_FAILURE_INJECTOR
 
+class SelectEvacFailureRegionClosure : public HeapRegionClosure {
+  CHeapBitMap& _regions;
+  size_t _evac_failure_regions_num;
+
+public:
+  SelectEvacFailureRegionClosure(CHeapBitMap& regions, size_t cset_length) :
+    _regions(regions),
+    _evac_failure_regions_num(cset_length * G1EvacuationFailureALotCSetPercent / 100) { }
+
+  bool do_heap_region(HeapRegion* r) override {
+    assert(r->in_collection_set(), "must be");
+    if (_evac_failure_regions_num > 0) {
+      _regions.set_bit(r->hrm_index());
+      --_evac_failure_regions_num;
+      return false;
```

Comment:
Just an initial comment to think about and discuss: This would most likely always ever select the same regions (probably eden) that are first in the collection set.
I think selecting by probability (with G1EvacuationFailureALotCSetPercent uniform probability) would be more interesting even if it makes the selection more random.

Reply:
Yes, currently I only make it select regions starting from first.
The reason I don't add random selection is that, currently we need this functionality to verify the functionality and perf improvement of JDK-8256265 implementation, so a more stable selection of evacuation failure regions will help to verify the effect of JDK-8256265.

I agree that random selection is an interesting option to supply too, I just file https://bugs.openjdk.java.net/browse/JDK-8277851 to track the issue.

Reply:
This PR makes the first X% regions in cset evac-fail, but `G1EvacuationFailureALotCSetPercent` doesn't show such bias. (It kind of implies uniform distribution, IMO.) It would be nice if the name and/or the explanation string could be more specific.